runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -83,6 +86,11 @@ def __init__(
         self.address = None
         self.client = None
 
+        # K8s related fields
```

Comment:
what was the reasoning for fully integrating kubernetes into ondemand cluster? code snippets like this/in the config_for_rns with k8s specific variables or k8s specific branches make it seem like it makes more sense and also pretty straightforward to have KubernetesCluster be a separate class but subclass OnDemandCluster (but could still reuse the same ondemand_cluster factory function).

would not really be user facing though so we could always make this refactor in a separate following PR if this makes sense.